src/main/java/com/exadel/discount/service/impl/JwtExtractionServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package com.exadel.discount.service.impl;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.exadel.discount.service.JwtExtractionService;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.Date;
+
+@Service
+@Setter
+@Slf4j
+public class JwtExtractionServiceImpl implements JwtExtractionService {
+    private final String ROLES_CLAIM_NAME = "role";
+
+    @Value("${jwt.encryption.key}")
+    private String TOKEN_ENCRYPTION_KEY;
+
+    @Override
+    public String getSubject(String token) {
+        log.debug("getting a subject from a token");
+        return JWT.require(Algorithm.HMAC256(TOKEN_ENCRYPTION_KEY))
+                .build().verify(token)
+                .getSubject();
+    }
+
+    @Override
+    public Date getExpiration(String token) {
```

Comment:
why do you need this expiration if .verify method already check expiration?
![image](https://user-images.githubusercontent.com/3354988/123936626-082a9280-d99e-11eb-8e82-1c3e5e2b331a.png)

Reply:
This is done locally.

Reply:
It was resolved in commit "Commit with amendments №2".